core/src/main/scala/org/apache/spark/storage/memory/MemoryStore.scala

```diff
@@ -325,6 +325,10 @@ private[spark] class MemoryStore(
 
     // Whether there is still enough memory for us to continue unrolling this block
     var keepUnrolling = true
+    // Number of elements unrolled so far
+    var elementsUnrolled = 0L
+    // How often to check whether we need to request more memory
+    val memoryCheckPeriod = 16
```

Comment:
Should this be made configurable? Or can we make sure it's the best magic number that we can choose?

Reply:
it's hard coded in `putIteratorAsValues` too, we can improve it later.

Reply:
I have just made it configurable. I'm not sure if this writting is reasonable.